javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java

```diff
@@ -17,69 +17,78 @@
+            "    public void chokes3() {\n" +
+            "        HairType<?> hairType = sheep.getHairType();\n" +
+            "        hairType.getRenderer().renderHair(sheep.getHairType(), sheep);\n" +
+            "        hairType.getRenderer();\n" +
+
+            "    }\n" +
+            "    \n" +
+            "    public void chokes() {\n" +
+            "        sheep.getHairType().getRenderer().renderHair(sheep.getHairType(), sheep);\n" +
+            "    }\n" +
+            "    \n" +
+            "    public void chokes2() {\n" +
+            "        HairType<?> hairType = sheep.getHairType();\n" +
+            "        hairType.getRenderer().renderHair(hairType, sheep);\n" +
+            "    }\n" +
+            "}";
+
+    // Expected Result MethodCallExpr in parsed code
+    private final static Map<String,String> resultsQualifiedName = new HashMap<>();
+
+    private final static Map<String,String> resultsResolvedType = new HashMap<>();
+
+    @BeforeAll
+    static void init() {
         resultsQualifiedName.put("sheep.getHairType()", "issue1945.interfaces.HairyAnimal.getHairType");
         resultsQualifiedName.put("hairType.getRenderer().renderHair(sheep.getHairType(), sheep)", "issue1945.interfaces.HairTypeRenderer.renderHair");
         resultsQualifiedName.put("hairType.getRenderer()", "issue1945.interfaces.HairType.getRenderer");
         resultsQualifiedName.put("sheep.getHairType().getRenderer().renderHair(sheep.getHairType(), sheep)", "issue1945.interfaces.HairTypeRenderer.renderHair");
         resultsQualifiedName.put("sheep.getHairType().getRenderer()", "issue1945.interfaces.HairType.getRenderer");
         resultsQualifiedName.put("hairType.getRenderer().renderHair(hairType, sheep)", "issue1945.interfaces.HairTypeRenderer.renderHair");
-        Map<String,String> resultsResolvedType = new HashMap<>();
+
         resultsResolvedType.put("sheep.getHairType()", "issue1945.interfaces.HairType<?>");
         resultsResolvedType.put("hairType.getRenderer().renderHair(sheep.getHairType(), sheep)", "void");
         resultsResolvedType.put("hairType.getRenderer()", "R");
         resultsResolvedType.put("sheep.getHairType().getRenderer().renderHair(sheep.getHairType(), sheep)", "void");
         resultsResolvedType.put("sheep.getHairType().getRenderer()", "R");
         resultsResolvedType.put("hairType.getRenderer().renderHair(hairType, sheep)", "void");
+    }
 
-        CompilationUnit cu = StaticJavaParser.parse(code);
-        List<MethodCallExpr> nodes = cu.findAll(MethodCallExpr.class);
-        for (MethodCallExpr expr : nodes) {
-            String qName = expr.resolve().getQualifiedName();
-            String resolvedType = expr.calculateResolvedType().describe();
-            assertEquals(resultsQualifiedName.get(expr.toString()), qName);
-            assertEquals(resultsResolvedType.get(expr.toString()), resolvedType);
-        }
+    private static List<MethodCallExpr> parsedCodeMethodCalls() {
```

Comment:
Why are you declaring this method static?

Reply:
It is our understanding that the factory method that provides inputs to a test must be declared static as per the [JUnit documentation](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource).